include/oneapi/tbb/parallel_sort.h

```diff
@@ -240,16 +246,25 @@ void parallel_sort( RandomAccessIterator begin, RandomAccessIterator end, const
 /** @ingroup algorithms **/
 template<typename RandomAccessIterator>
     __TBB_requires(std::random_access_iterator<RandomAccessIterator> &&
-                   less_than_comparable<typename std::iterator_traits<RandomAccessIterator>::value_type>)
+                   less_than_comparable<iter_value_type<RandomAccessIterator>> &&
+                   std::swappable<iter_value_type<RandomAccessIterator>> &&
+                   std::move_constructible<iter_value_type<RandomAccessIterator>> &&
+                   std::assignable_from<iter_value_type<RandomAccessIterator>&, iter_value_type<RandomAccessIterator>>)
 void parallel_sort( RandomAccessIterator begin, RandomAccessIterator end ) {
     parallel_sort(begin, end, std::less<typename std::iterator_traits<RandomAccessIterator>::value_type>());
 }
 
+template<typename Range>
+using range_value_type = typename std::iterator_traits<range_iterator_type<Range>>::value_type;
```

Comment:
May be we can move this helper to `_range_common.h` and simplify `tbb::parallel_for_each` constraints? The same for `iter_value_type`

Reply:
I don't think so, As I know the `parallel_for_each` algorithm checks the requirements on reference types instead of value types.